### PR TITLE
[Parvathy, Rahul] | BAH-3068 | Add. Shell Script To Run Latest Images

### PR DESCRIPTION
**Pull request by rahu1ramesh**
JIRA -> [BAH-3068](https://bahmni.atlassian.net/browse/BAH-3068)

This PR enhances the run-bahmni.sh script to take an optional param which can be used to pass a different .env file, for instance the `.env.dev` file which mentions all images tagged as `latest` for development.

The updated [`README.md`](https://github.com/Bahmni/bahmni-docker/blob/418ba10af791d1c1837c4768bd49b9a4aa115b7b/README.md) provides clear instructions on how to choose the appropriate script based on the desired image versions and how to configure the respective `.env` or `.env-dev` files before running the commands.

### CODE REVIEW

**Comment by gsluthra** [comment]
As discussed, recommend to **not** introduce a new run-bahmni-dev.sh. Instead use an ENV variable setting to decide if `run-bahmni.sh` picks `.env` file or `.env.dev` or some other env file. e.g: `RUN_BAHMNI_DOCKER_ENV_FILE=.env.dev`

**Comment by rahu1ramesh** [comment]
> You can also pass a parameter to the script. `./runbahmni.sh .env` or `./runbahmni.sh .env.dev`. Default to .env when no parameter is passed

@gsluthra Passing a parameter seems like an easier fix than having to introduce a new ENV variable. This just needs to be updated in the README.md file. Any suggestions?

File: run-bahmni.sh [hunk 1]
```diff
@@ -45,32 +45,40 @@ function checkIfDirectoryIsCorrect {
 }
 
 function start {
-    echo "Executing command: 'docker compose up -d'"
-    echo "Starting Bahmni with default profile from .env file"
-    docker compose up -d
+    if [ "$1" == ".env" ]; then
```
Comment (gsluthra) [inline run-bahmni.sh:48]: No. I don't like this. People should not need to pass `.env` -- that is the default for docker. People should ONLY pass a different .env if they want something besides default. If they pass a param, then that is used for starting the environment, if they don't it remains as `.env` by default.

File: run-bahmni.sh [hunk 2]
```diff
@@ -86,18 +94,25 @@ function showOpenMRSlogs {
...
+    docker compose --env-file "$1" --profile bahmni-mart up -d
 }
 
 function pullLatestImages {
-    echo "Pulling all latest images..."
-    docker compose pull
+    if [ "$1" == ".env" ]; then
+        echo "Pulling all 1.0.0 images..."
+    elif [ "$1" == ".env.dev" ]; then
+        echo "Pulling all latest images..."
```
Comment (gsluthra) [inline run-bahmni.sh:104]: We don't know this for sure. People can change .env or .env.dev to contain different images. Just provide a console message saying: Running with which .env file. That is all we know for sure.

**Review by gsluthra — CHANGES_REQUESTED** [review]

File: run-bahmni.sh [hunk 3]
```diff
@@ -45,40 +45,38 @@ function checkIfDirectoryIsCorrect {
 }
 
 function start {
-    if [ "$1" == ".env" ]; then
-        echo "Executing command: 'docker compose up -d' with 1.0.0 images"
-        echo "Starting Bahmni with default profile from .env file"
-    elif [ "$1" == ".env.dev" ]; then
+    if [ "$file" == ".env.dev" ]; then
```
Comment (gsluthra) [inline run-bahmni.sh:48]: This should just be $file is not empty. If not empty, I will pass it as-is. Someone can create a file called: env.production or env.litev1 etc and pass that to run-bahmni.sh. So, the check should just be, if some param is passed, we will run the --env-file with that param. Else we will run without any --env-file.

File: run-bahmni.sh [hunk 4]
```diff
@@ -94,32 +92,30 @@ function showOpenMRSlogs {
...
-    if [ "$1" == ".env" ]; then
-        echo "Pulling all 1.0.0 images..."
-    elif [ "$1" == ".env.dev" ]; then
+    if [ "$file" == ".env.dev" ]; then
         echo "Pulling all latest images..."
+        docker compose --env-file "$file" pull
     else
-        echo "[ERROR] Incorrect Environment Variable file path. Please provide either .env or .env.dev file as argument."
-        exit 1
+        echo "Pulling all 1.0.0 images..."
```
Comment (gsluthra) [inline run-bahmni.sh:103]: We don't know this for sure if it is 1.0.0 or not. Just say: "Pulling all images ..."

**Comment by gsluthra** [comment]
Please REMOVE all mention of .env.dev from the run-bahmni.sh script. People should be able to pass ANY env file. The script should not assume that passed file is .env or something else. Someone can create `.env.production` and pass that. Or they can create `.env.lite-v1` and pass that. We can only assume that either they pass no file, or they pass _some_ file. If they pass no file, we run the command as-is (without passing any --env-file). If they pass a file, we run all docker command with --env-file option. 

No other change in the run-bahmni.sh script.

File: run-bahmni.sh [hunk 5]
```diff
@@ -45,40 +45,38 @@ function checkIfDirectoryIsCorrect {
...
-        echo "Starting Bahmni with default profile from .env file"
-    elif [ "$1" == ".env.dev" ]; then
+    if [ "$file" == ".env.dev" ]; then
         echo "Executing command: 'docker compose up -d' with latest images"
         echo "Starting Bahmni with default profile from .env.dev file"
+        docker compose --env-file "$file" up -d
     else
-        echo "[ERROR] Incorrect Environment Variable file path. Please provide either .env or .env.dev file as argument."
-        exit 1
+        echo "Executing command: 'docker compose up -d' with 1.0.0 images"
```
[the remark this answers is not in the document]
Comment (rahu1ramesh) [reply run-bahmni.sh]: Added this message instead 
```
echo "Executing command: 'docker compose up -d' with the images specified in the $file file"
echo "Starting Bahmni with default profile from $file file"
```

File: run-bahmni.sh (hunk 3 above)
Comment (rahu1ramesh) [reply run-bahmni.sh]: We have now set `.env` as the default and will take in any argument unless it's empty.
```
file=".env"
if ! [ "$1" == "" ]; then
    file="$1"
fi
```

File: run-bahmni.sh (hunk 2 above)
Comment (rahu1ramesh) [reply run-bahmni.sh]: Updated the message
`echo "Pulling all the images specified in the $file file..."`

File: README.md [hunk 6]
```diff
@@ -2,7 +2,24 @@
...
-1. Go to bahmni-lite or bahmni-standard subfolder. For example: `cd bahmni-lite`.
-2. Execute script: `./run-bahmni.sh`. This will give you options for start/stop/view-logs/pull/reset/etc. 
-3. Ensure your `.env` file in the sub-folder has correct PROFILE configured, before executing the above commands.  
+## Running Bahmni LITE or STANDARD using docker compose: 
+1. Navigate to the relevant subfolder for your desired configuration. For example: `cd bahmni-lite`.
+2. Execute the script: `./run-bahmni.sh`. This script provides various options such as start, stop, view logs, pull updates, reset, etc.
+3. Before executing the above commands, ensure that your `.env` file in the sub-folder is correctly configured with the appropriate PROFILE.
+
+## Environment Variable Configuration For Bahmni Lite
+The `.env` and `.env.dev` files are used for configuring environment variables for the Bahmni Lite Docker setup. The `.env` file points to the `1.0.0` image tag, while `.env.dev` points to the `latest` image.
```
Comment (gsluthra) [inline README.md:11]: The 1.0.0 images are the stable tested version of Bahmni Lite v1.0, and we recommend you run with these images for production purpose.

File: README.md [hunk 7]
```diff
@@ -2,7 +2,24 @@
...
+
+By default `run-bahmni.sh` script runs with the `.env`, that uses the `1.0.0` images
+```shell
```
Comment (gsluthra) [inline README.md:14]: If you want to run `docker compose` commands directly, then use the `--env-file` option to pass this file. e.g: `docker compose up --env-file .env.dev`

File: README.md [hunk 8]
```diff
@@ -2,7 +2,24 @@
...
+run-bahmni.sh
+```
+
+- Instead if you wish to use the `latest` images, run the `run-bahmni.sh` script with the argument `.env.dev`
+```shell
+run-bahmni.sh .env.dev
+```
+
```
Comment (gsluthra) [inline README.md:22]: If you want to run docker compose commands directly, then use the `--env-file` option to pass this file. e.g: `docker compose up --env-file .env.dev`

File: run-bahmni.sh [hunk 9]
```diff
@@ -86,25 +86,25 @@ function showOpenMRSlogs {
...
-    docker compose --profile bahmni-lite --profile bahmni-standard --profile bahmni-mart ps
+    docker compose --env-file "$file" --profile bahmni-lite --profile bahmni-standard --profile bahmni-mart ps
 
 }
 
 
 # Function to prompt the user for a "Yes" or "No" answer
 confirm() {
-    read -p "$1 [y/n]: " response
+    read -p "$file [y/n]: " response
```
Comment (gsluthra) [inline run-bahmni.sh:107]: Why is this changed to $file?

File: run-bahmni.sh [hunk 10]
```diff
@@ -157,14 +157,14 @@ function resetAndEraseALLVolumes {
...
 
     echo "Enter the name of the SERVICE to restart:"
     read serviceName
     
     echo "Restarting SERVICE: $serviceName"
-    docker compose restart $serviceName
+    docker compose --env-file "$file" restart $serviceName
 
     if confirm "Do you want to see the service logs?"; then
         docker compose logs $serviceName -f
```
Comment (gsluthra) [inline run-bahmni.sh:170]: Doesn't this command need --env-file option?

File: run-bahmni.sh [hunk 11]
```diff
@@ -157,14 +157,14 @@ function resetAndEraseALLVolumes {
 
 function restartService {
     # One can ONLY restart services in current profile (limitation of docker compose restart command). 
-    echo "Listing the running services from current profile (.env file) that can be restarted..."
+    echo "Listing the running services from current profile ($file file) that can be restarted..."
     docker compose ps
```
Comment (gsluthra) [inline run-bahmni.sh:161]: Doesn't the ps command need --env-file flag?

File: run-bahmni.sh (hunk 9 above)
Comment (rahu1ramesh) [reply run-bahmni.sh:107]: This was done by mistake, will revert this.

File: README.md (hunk 6 above)
Comment (rahu1ramesh) [reply README.md:11]: Will update the README.md doc.

File: run-bahmni.sh (hunk 10 above)
Comment (rahu1ramesh) [reply run-bahmni.sh]: Added file path option.

File: README.md (hunk 8 above)
Comment (rahu1ramesh) [reply README.md:30]: Added to the README.md.

**Review by gsluthra — APPROVED** [review]
